code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/RegularDeploymentUnitInstance.java

```diff
@@ -0,0 +1,173 @@
+package io.cattle.platform.servicediscovery.deployment.impl;
+
+import static io.cattle.platform.core.model.tables.ImageTable.IMAGE;
+import static io.cattle.platform.core.model.tables.InstanceTable.INSTANCE;
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.model.Image;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.core.model.ServiceExposeMap;
+import io.cattle.platform.deferred.util.DeferredUtils;
+import io.cattle.platform.engine.process.impl.ProcessCancelException;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.cattle.platform.object.process.StandardProcess;
+import io.cattle.platform.object.resource.ResourceMonitor;
+import io.cattle.platform.object.resource.ResourcePredicate;
+import io.cattle.platform.servicediscovery.api.dao.ServiceExposeMapDao;
+import io.cattle.platform.servicediscovery.deployment.DeploymentUnitInstance;
+import io.cattle.platform.servicediscovery.resource.ServiceDiscoveryConfigItem;
+import io.cattle.platform.servicediscovery.service.ServiceDiscoveryService;
+import io.cattle.platform.storage.service.StorageService;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+public class RegularDeploymentUnitInstance extends DeploymentUnitInstance {
```

Comment:
nit: Maybe `DefaultDeploymentUnitInstance` would be a better name.  Who's to say what is regular or not.  It just feels this may seem unkind if you also create a `StarBellySneetchDeploymentUnitInstance` who will then feel as though he is not as important as the `RegularDeploymentUnitInstance` manager.